src/comm/TcpAcceptor.cc

```diff
@@ -362,6 +331,10 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
         if (ignoreErrno(errcode) || errcode == ECONNABORTED) {
             debugs(50, 5, status() << ": " << xstrerr(errcode));
             return Comm::NOMESSAGE;
+        } else if (errcode == EBADF) {
+            debugs(50, 2, status() << ": " << xstrerr(errcode));
+            mustStop("Listening socket closed");
+            return Comm::NOMESSAGE;
```

Comment:
Because on shutdown/reconfigure TcpAcceptor is *not* the class/code closing the socket. In those events there is a race between read and close AsyncCall scheduling. We discover the closure through accept(2) return values when the read signal/Call wins the race.

Reply:
Once ERR_CLOSING flag was handled by the acceptOne() this turned out to be unnecessary. Removed by c92b96b.